pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -2963,6 +2961,19 @@ public String startReplaceSegments(String tableNameWithType, List<String> segmen
     return segmentLineageEntryId;
   }
 
+  // TODO: Add more conflict checks over segmentsTo later.
+  private static boolean isConflicted(List<String> segmentsFrom, LineageEntry lineageEntry, TableConfig tableConfig) {
+    if (isRefreshTable(tableConfig)) {
+      return CollectionUtils.isEqualCollection(segmentsFrom, lineageEntry.getSegmentsFrom());
+    }
+    // For APPEND table, skip conflict checks if segmentsFrom is empty.
```

Comment:
Add TODO here saying for APPEND table, if the new segments from 2 batch jobs are overlapping, then we should reject either one of the batch jobs.